### PR TITLE
5 boosting algorithms

### PR DESCRIPTION
**Pull request by KseniiaSych**

### CODE REVIEW

File: 2_decision_tree_gridsearch/Boosting_algorithms.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    features = df.drop(tagetName, axis=1)
+    return features, target
+
+def balance_undersample(df, tagetName='Survived'):
+    survived_part = df[df[tagetName]==1]
+    not_survived_sample = df[df[tagetName]==0].sample(n=len(survived_part))
+    data =  pd.concat([survived_part, not_survived_sample], ignore_index=True)
+    return data
+
+def remove_over_95th(df, columns=[]):
```
Comment (milkisbad) [inline 2_decision_tree_gridsearch/Boosting_algorithms.md]: Names in this function. 95 and .98 and .2 don't make sense, plus 'copy' is bad name for 'df_copy' or 'filtered_df'

File: 2_decision_tree_gridsearch/Boosting_algorithms.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+```
+
+# XGBoost
+
+```python
+xgd_clf = XGBClassifier(use_label_encoder=False, eval_metric='mlogloss')
+```
+
+```python
+xgd_clf.fit(X_preprocessed, Y)
```
Comment (milkisbad) [inline 2_decision_tree_gridsearch/Boosting_algorithms.md]: missing gridsearch here

File: 2_decision_tree_gridsearch/Boosting_algorithms.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+grad_clf.score(x_test_processed, y_test)
+```
+
+```python
+y_pred = grad_clf.predict(x_test_processed)
+print_scores(y_test, y_pred)
+```
+
+```python
+grad_param_grid = dict(
```
Comment (milkisbad) [inline 2_decision_tree_gridsearch/Boosting_algorithms.md:186]: the grid has weird param values. What you can try is having the grid of only the most influencial - `learning_rate, max_depth, n_estimators` and do in depth analysis of them (starting from values that would be obviously too low. check this for reference. https://medium.com/all-things-ai/in-depth-parameter-tuning-for-gradient-boosting-3363992e9bae .
